### PR TITLE
committing initial image func

### PR DESCRIPTION
**Pull request by cooktheryan**
fix of the target name

just write the image to memory

CI testing

Signed-off-by: Ryan Cook <rcook@redhat.com>

### CODE REVIEW

**Comment by josephsawaya** [comment]
if `Image.Process` never calls `currentToLatest` then the `Apply` and `MethodEngine` aren't getting called either so if you just want `Process` to call `loadPodman` every time regardless of what changes have been made then you can implement `MethodEngine` and `Apply` as no-ops

File: pkg/engine/types.go [hunk 1]
```diff
@@ -27,7 +27,6 @@ type FetchitConfig struct {
 	TargetConfigs []*TargetConfig `mapstructure:"targetConfigs"`
 	ConfigReload  *ConfigReload   `mapstructure:"configReload"`
 	PAT           string          `mapstructure:"pat"`
-	volume        string          `mapstructure:"volume"`
```
Comment (sallyom) [inline pkg/engine/types.go:30]: nice catch, cleanup!

File: pkg/engine/image.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+func (i *Image) GetKind() string {
+	return imageMethod
+}
+
+func (i *Image) SchedInfo() SchedInfo {
+	return SchedInfo{
+		schedule: i.Schedule,
+		skew:     i.Skew,
+	}
+}
```
Comment (sallyom) [inline pkg/engine/image.go:32]: The individual SchedInfo fn isn't necessary, you can remove that, https://github.com/containers/fetchit/pull/194  (SchedInfo is the same for all targets, so it's a method of the CommonMethod type only, and not the interface)

**Comment by sallyom** [comment]
@cooktheryan you can add this type to the comment here - to keep track - we might add these to a slice later, rather than check `if target.url != ""`   https://github.com/containers/fetchit/blob/main/pkg/engine/fetchit.go#L279  - we'll have Clean, ConfigTarget, & Image that don't use git in processing. (& systemd.AutoUpdateAll)

File: pkg/engine/image.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+func (i *Image) Apply(ctx, conn context.Context, currentState, desiredState plumbing.Hash, tags *[]string) error {
+	changeMap, err := applyChanges(ctx, i.GetTarget(), i.GetTargetPath(), currentState, desiredState, tags)
+	if err != nil {
+		return err
+	}
+	if err := runChangesConcurrent(ctx, conn, i, changeMap); err != nil {
+		return err
+	}
+	return nil
+}
```
Comment (sallyom) [inline pkg/engine/image.go:60]: can return nil for these, like the Clean method does (like @josephsawaya mentioned): https://github.com/containers/fetchit/blob/main/pkg/engine/clean.go#L42-#L67

**Pull request merged**